Tests/NetworkServiceTests/NetworkServiceTests.swift

```diff
@@ -1,8 +1,114 @@
-import NetworkService
+@testable import NetworkService
 import XCTest
 
 final class NetworkServiceTests: XCTestCase {
-    func testUseMethodExistsInAPI() {
-        NetworkService().testUse()
+    struct MockEndpoint: HTTPAPIEndpoint {
+        static var networkEnvironment: String = "test"
+        var baseURL: URL = .init(string: "https://api.example.com")!
+        var path: String = "/endpoint"
+        var task: HTTPTask = .request
+        var method: HTTPMethod = .get
+        var headers: HTTPHeaders = [:]
+    }
+
+    var networkService: NetworkService<MockEndpoint>!
+
+    override func setUpWithError() throws {
+        networkService = NetworkService()
+        URLProtocol.registerClass(MockURLProtocol.self)
+    }
+
+    override func tearDownWithError() throws {
+        URLProtocol.unregisterClass(MockURLProtocol.self)
+        networkService = nil
+    }
+
+    func testSuccessfulRequestReturnsExpectedData() async throws {
+        // Given
+        let expectedData = "Mocked Response Data".data(using: .utf8)!
+        MockURLProtocol.requestHandler = { request in
+            let response = HTTPURLResponse(url: request.url!, statusCode: 200, httpVersion: nil, headerFields: nil)!
+            return (response, expectedData)
+        }
+
+        let endpoint = MockEndpoint()
+
+        // When
+        let result = await networkService.request(endpoint)
+
+        // Then
+        switch result {
+        case let .success(data):
+            XCTAssertEqual(data, expectedData)
+        case let .failure(error):
+            XCTFail("Expected success, but got failure: \(error)")
+        }
+    }
+
+    func testBadRequestReturnsFailure() async throws {
```

Comment:
I think this name explains more clearly the intention of the test.
```suggestion
    func testBadRequestStatusCodeInResponseTriggersBadRequestError() async throws {
```

Reply:
Agree, fixed with 203e327